components/GameForm.tsx

```diff
@@ -37,34 +46,53 @@ function GameForm(props) {
     reValidateMode: "onChange",
   });
 
+  if (gameId) {
```

Comment:
you should never ever have conditional hooks, no matter what, the component should always use the same hooks for each render, otherwise you'll lose control on what is happening in that component.

So no wrapping hooks with `if` or using `return` before all the hooks declarations